src/eos-autoupdater.c

```diff
@@ -599,10 +601,29 @@ is_time_to_update (guint update_interval_days,
      * computer’s run eos-autoupdater. In order to avoid a thundering herd of
      * computers requesting updates when a lab is first turned on, create a
      * stamp file with a random delay applied, and check again for updates
-     * later. */
-    g_debug ("Not time to update, due to stamp file not being present.");
-    update_stamp_file (update_interval_days, randomized_delay_days);
-    is_time_to_update = FALSE;
+     * later. To do this, we fake the date of the most recent successful update
+     * to be @update_interval_days in the past, so only the
+     * @randomized_delay_days is taken into account for triggering the next
+     * update. */
+    if (randomized_delay_days > 0)
+      {
+        guint64 last_successful_update_secs;
+
+        g_debug ("Not time to update, due to stamp file not being present, but %s is set to %u days.",
+                 RANDOMIZED_DELAY_KEY, randomized_delay_days);
+        last_successful_update_secs = g_get_real_time () / G_USEC_PER_SEC;
+        if (last_successful_update_secs >= update_interval_days * SEC_PER_DAY)
+          last_successful_update_secs -= update_interval_days * SEC_PER_DAY;
```

Comment:
> If you deem this scenario to be too strained, then fine, the code looks ok

I think that scenario is too strained. If the user edits the configuration, any changes to make the `IntervalDays` shorter will result in an update being correctly run the next time `eos-autoupdater` is triggered. Any edits to make it longer will result in an additional delay before the first update, as you say, but that’s still in the same direction as what the user wants (delaying updates for a longer period). It’s unlikely to impact on dev testing, since `IntervalDays` is normally edited down to `0` (not up) by developers testing `eos-updater`. I don’t think the extra complexity is worth it.